### ISSUE TITLE
new rule - detecting string enumerations that look like string concatenations

### ISSUE THREAD

**Issue by spaceby** [astral-sh/ruff#13534]
```python
a = ('Hello ',
     + 'world')
```
This looks like an explicit string concatenation, but it's actually an enumeration of strings.
I wish ruff would detect this

It seems it should work when there is the following configuration

```toml
[tool.ruff.lint.flake8-implicit-str-concat]
allow-multiline = false
```

**Comment by MichaReiser** [comment]
That's probably something a type checker can find very easily because the "+" operator isn't defined for string. Overall, I would prefer to implement this as a more generic rule where `+` is used on an operand that doesn't support it rather than special casing it to strings.

**Comment by AlexWaygood** [comment]
I agree with @MichaReiser. This is something that is already implemented by existing Python type checkers and will be supported by red-knot when it's ready. I think it's best for us to leave it for now in Ruff. But thanks for the suggestion!
